### ISSUE TITLE
Import trip data and charge data to Home Assisstant

### ISSUE THREAD

**Issue by lucasimons** [flobz/psa_car_controller#356]
Hi, I would like to import the data of the charging sessions and trips, I have installed both influx and grafana but I cannot read the data I have installed everything as a Home Assistant addon

**Environment **
 - OS: HassOS
 - Peugeot e208

**Additional context**

![image](https://user-images.githubusercontent.com/67467751/165346199-15c0b719-5bad-4a9f-8f47-e94abb1b72e7.png)

**Comment by kurim** [comment]
Hi lucasimons,

did you enable the external port? To open them in influx/grafana.
![image](https://user-images.githubusercontent.com/1880220/165596512-1f4fd789-7466-4c6c-9efc-f84188ca8291.png)

If you want transfer data from other source like manual installation to HASS it is quite complicated, because you need to replace the .db file to get those data in HASS.

**Comment by lucasimons** [comment]
I have enabled the port in fact I can see the data but I don't know how to import the trip and charge data inside Home Assistant as sensors .
I wanted to create a sensor that counted how many kWh I lcharged and how many kilometers I did, for example, monthly or daily

**Comment by kurim** [comment]
Ok~
you can use node red to import the data for example.
This is what I do, it is more flexible then the other options like creating sensors in config

**Comment by lucasimons** [comment]
Did you manage to read the charge data via node red?  For example, how much energy did you charge in the last session?  Could I share the nodes?

**Comment by kurim** [comment]
Good point, no that's nothing I test so far, since i just use this API: /get_vehicleinfo/ to read those data you see.

Here is my NodeRedFlow: [flows.zip](https://github.com/flobz/psa_car_controller/files/8579978/flows.zip)

About calculating charge value for charge session, I think you can do it like this:
Check when Charge start + current SOC and check charge end then calculate the value between soc start+end.
or flobz need to create an api which provide those data as well.

**Comment by lucasimons** [comment]
thanks 

this is what that i would to do...

https://grafana.szambetti.it/d/CPrt7kvnk/e208?orgId=1&kiosk=tv 

@flobz is it possible with psa car controller installe as addon?

**Comment by kurim** [comment]
You can analyse the panels, as far as I can see they do some SQL query's to get the data.

```sql
SELECT
    strftime('%Y-%m-%dT%H:%M:%S', start_at) || "Z00:00" AS start_time,
    COUNT(*) AS times,
    SUM((end_level - start_level) * .42) AS kwh_charged,
    SUM(((end_level - start_level) * .42) / ((strftime('%s', stop_at) - strftime('%s', start_at)) / 3600.0000))
	COUNT(*) AS average_kw
FROM
    battery
WHERE
    ((end_level - start_level) * .42) /((strftime('%s', stop_at) - strftime('%s', start_at)) / 3600.0000) < 22
```

Many things you might be able to do with NodRed as well, but you may need a database to store the values.
But this is a different approach.

**Comment by kurim** [comment]
Btw. why not follow this: https://github.com/flobz/psa_car_controller/issues/161
Since it is just another install way

**Comment by lucasimons** [comment]
Because I have only a rpi with HassOS installed and I works fine, I doesn't have a Linux system to use this method...

**Comment by kurim** [comment]
the problem is you don't have telegraf which seems to be essential for data transfer.
For me i have it done by rednode+mysql:
![image](https://user-images.githubusercontent.com/1880220/166144823-6b651bf6-bbef-46b9-9ae4-a519e13a45b5.png)

**Comment by kurim** [comment]
Hello again,

I have test it with: HassOS
I have installed:
- PSA 3.0.3
- InfluxDB (https://addons.community)
- Telegraf (https://github.com/Sabuto/hassio-repo)
- Grafana (https://addons.community)

After install everything you need to change the configs:

InfluxDB:
```yaml
auth: false
certfile: fullchain.pem
envvars:
  - name: INFLUXDB_HTTP_LOG_ENABLED
    value: 'true'
keyfile: privkey.pem
reporting: true
ssl: true
log_level: info
```
Telegraf:
```yaml
custom_conf:
  enabled: true
  location: /config/telegraf.conf
```
Rest of this config can be ignored

After this open [File Editor] or [VSCode] and create the telegraf.conf in conf folder. (like mentioned in Grafana Guide)
![image](https://user-images.githubusercontent.com/1880220/166159099-daca9c0c-a1f0-466c-b382-7a39a5851dd1.png)

After this step you can Start InfluxDB + Telegraf + Grafana.
Everytime you need to wait till everything loads complete (first time it take a while).
Now open Grafana and use instead of localhost the fqdn of you device like: http://homeassistant.fritz.box:8086
And now you can create panels like in your example:
![image](https://user-images.githubusercontent.com/1880220/166159266-a5d83513-a6be-414b-a647-69812c897255.png)

**Comment by stevoh6** [comment]
> You can analyse the panels, as far as I can see they do some SQL query's to get the data.
> 
> ```sql
> SELECT
>     strftime('%Y-%m-%dT%H:%M:%S', start_at) || "Z00:00" AS start_time,
>     COUNT(*) AS times,
>     SUM((end_level - start_level) * .42) AS kwh_charged,
>     SUM(((end_level - start_level) * .42) / ((strftime('%s', stop_at) - strftime('%s', start_at)) / 3600.0000))
> 	COUNT(*) AS average_kw
> FROM
>     battery
> WHERE
>     ((end_level - start_level) * .42) /((strftime('%s', stop_at) - strftime('%s', start_at)) / 3600.0000) < 22
> ```
> 
> Many things you might be able to do with NodRed as well, but you may need a database to store the values. But this is a different approach.
> 
> Notice: 42 seems not correct because battery has 46kWh (for Corsa-E) instead of 42kWh

HI, could you please explain the where condition? Why 22? thx :-)

**Comment by lucasimons** [comment]
> ```yaml
> custom_conf:
>   enabled: true
>   location: /config/telegraf.conf
> ```

Can you import the trip / charge session to influx?
